crates/driver/src/commit_reveal.rs

```diff
@@ -145,6 +150,46 @@ impl CommitRevealSolving for CommitRevealSolver {
             )),
         }
     }
+
+    fn account(&self) -> &Account {
+        self.solver.account()
+    }
+
+    fn name(&self) -> &str {
+        self.solver.name()
+    }
+}
+
+/// This is just a wrapper type to make a `dyn CommitRevealSolving` usable where `dyn Solver` is
+/// expected. This type is only supposed to give information about the name and account of the
+/// underlying solver and will panic if `solve()` gets called.
+#[derive(Clone)]
+pub struct CommitRevealSolverAdapter {
```

Comment:
uber-naming-nit: I would include in the name somewhere that its meant as a way of adapting a `CommitRevealSolver` for legacy components:
```suggestion
pub struct CommitRevealLegacySolverAdapter {
```

Not 100% sure this is a good idea, so feel free to ignore my suggestion.

Reply:
My understanding is that this will eventually be removed right?

Reply:
> My understanding is that this will eventually be removed right?

Yes. Eventually I would like to pass around something like
```rust
struct NamedAccount {
    name: String,
    account: Account
}
```
instead of `Arc<dyn Solver>` because that's the only data which is actually needed for all the logging related code.

Reply:
Regarding the naming nit: I'm not sure the `Legacy` part in the name gives us much so I would just be even more explicit in the doc comments for the type. I will also created this [issue](https://github.com/cowprotocol/services/issues/469) so I don't forget about the refactor.